### PR TITLE
Removing 'fields' from JSON output when the field is null.

### LINKED ISSUE #126: Null fields are exported
Using the default config file of Filebeat, where there are no fields defined, the exported fields are null.
Here is the JSON:

"_source": {
    "count": 1,
    "fields": null,
    "fileinfo": {},
    "line": 27320,
    "message": "\t    497799835,",
    "offset": 2768605,
    "shipper": "mar.localdomain",
    "source": "/var/log/commerce.log",
    "timestamp": "2015-10-21T21:15:34.590Z",
    "type": "log"
  },

### PR DESCRIPTION
**Pull request by andrewkroh**
Also added timestamp to FileEvent so that the event can be timestamp as soon as it is read.

Closes #126

### CODE REVIEW

File: input/file.go [hunk 1]
```diff
@@ -44,6 +47,24 @@ func (f *FileEvent) GetState() *FileState {
 	return state
 }
 
+func (f *FileEvent) ToMapStr() common.MapStr {
+	event := common.MapStr{
+		"timestamp": common.Time(f.Time),
```
Comment (ruflin) [inline input/file.go]: perhaps we should even call this line something like lineReadTimestamp or something similar to better describe what it is.

Comment (andrewkroh) [inline input/file.go]: Yes, that would be a better. I will make that change.

Comment (andrewkroh) [inline input/file.go]: How about shortened to `ReadTime`?

Comment (ruflin) [inline input/file.go]: I like that. I'm thinking if we could have something generic that is add to all beats events. Like beatCreateTime. In a way every message sent is a beat and that is the creation time of the beat :-)

**Comment by andrewkroh** [comment]
I renamed the field to `ReadTime` and also updated the CHANGELOG.md.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ All documentation about Filebeat can be found here.
 ### Backward Compatibility Breaks
 
 ### Bugfixes
+- Omit 'fields' from event JSON when null.
```
Comment (ruflin) [inline CHANGELOG.md]: Pull request id?

**Comment by ruflin** [comment]
@andrewkroh Can you update fields.yml with the new value?

File: CHANGELOG.md (hunk 2 above)
Comment (andrewkroh) [inline CHANGELOG.md]: Added.

**Comment by andrewkroh** [comment]
Regarding fields.yml, I updated the description of `timestamp`. Maybe we were talking about different things. Did you mean to rename the `timestamp` field present in the JSON? I was talking about renaming the `FileEvent.Time` field.

File: input/file.go [hunk 3]
```diff
@@ -44,6 +47,24 @@ func (f *FileEvent) GetState() *FileState {
 	return state
 }
 
+func (f *FileEvent) ToMapStr() common.MapStr {
```
Comment (urso) [inline input/file.go]: With all fields of FileEvent being visible I don't feel like this requires a method, but a function

Comment (ruflin) [inline input/file.go]: @urso It is definitively not required, but as it is fully related to FileEvent I actually quite like that it is a method.

File: input/file.go [hunk 4]
```diff
@@ -44,6 +47,24 @@ func (f *FileEvent) GetState() *FileState {
...
+	event := common.MapStr{
+		"timestamp": common.Time(f.ReadTime),
+		"source":    f.Source,
+		"offset":    f.Offset,
+		"line":      f.Line,
+		"message":   f.Text,
+		"fileinfo":  f.Fileinfo,
+		"type":      "log",
```
Comment (ruflin) [inline input/file.go]: @andrewkroh Just realised this line here is hardcoded. Should be dynamic, I opened and issue here for it: https://github.com/elastic/filebeat/issues/133

**Pull request merged**